app/services/candidates/registrations/behaviours/teaching_preference.rb

```diff
@@ -41,6 +42,14 @@ def get_available_subject_choices
         def all_available_subject_choices
           Candidates::School.subjects.map(&:last)
         end
+
+        def second_choice_different_than_first_choice
+          return if subject_second_choice.blank?
```

Comment:
Not a biggie, but you could also pass the condition in the validate method, like:

```
validate :second_choice_different_than_first_choice, if: -> { subject_second_choice.present? }
```

Reply:
Pushed this 👍